aquadoggo/src/validation.rs

```diff
@@ -86,44 +86,36 @@ pub async fn verify_log_id<S: StorageProvider>(
 /// This method determines the expected skiplink given an author, log id and sequence number. It
 /// _does not_ verify that this matches the skiplink encoded on any entry.
 ///
-/// If the expected skiplink target could not be found in the database an error is returned.
+/// An error is returned if:
+/// - seq num 1 was passed in, which can not have a skiplink
+/// - the expected skiplink target could not be found in the database.
 pub async fn get_expected_skiplink<S: StorageProvider>(
     store: &S,
     author: &Author,
     log_id: &LogId,
     seq_num: &SeqNum,
 ) -> Result<S::StorageEntry> {
-    // Derive the expected skiplink sequence number from the given sequence number.
     ensure!(
         !seq_num.is_first(),
         anyhow!("Entry with seq num 1 can not have skiplink")
     );
 
-    let expected_skiplink_seq_num = seq_num.skiplink_seq_num();
-    let expected_skiplink = match expected_skiplink_seq_num {
-        // Retrieve the expected skiplink from the database
-        Some(skiplink_seq_num) => {
-            store
-                .get_entry_at_seq_num(author, log_id, &skiplink_seq_num)
-                .await?
-        }
+    // Unwrap because method always returns `Some` for seq num > 1
```

Comment:
Depending on how we handle [this](https://github.com/p2panda/p2panda/issues/417#issuecomment-1203823769) this would have to be changed again later.